Tests/SwiftPhoenixClientTests/Serializer/SerializerTest.swift

```diff
@@ -91,37 +91,45 @@ final class SerializerTest {
         #expect(message.payloadString == "{\"foo\":\"bar\"}")
     }
     
+    private struct NumberMessage: Codable {
+        let int: Int
+        let float: Float
+    }
+    
     @Test func test_decodeMessageWithNumbers() throws {
         let text = """
-        ["1","2", "topic","event",{"int":1,"float":1.1}]
+        ["1","2", "topic","event",{"int":1234,"float":1.1234}]
```

Comment:
This was one of my issues using JSONSerialization, the loss of some precision of float values when serializing back and forth between data and text, and why I went down the JsonElement route. 

I don't have an answer right now but curious if you have any ideas.

Reply:
I changed these back. Definitely agree that the concern in round-trip encoding is that something changes!

But I'm not sure I understand the issue here -- do you have a test that shows loss of precision?

Reply:
When the test case is 1.1, the decoded message payload decodes into a `float: 1.1000002`. This wasn't the case with JsonElement

<img width="359" alt="Screenshot 2025-01-18 at 8 52 03 PM" src="https://github.com/user-attachments/assets/2aa30abb-ce38-43d5-97eb-d0cc0467fc60" />

Obviously this is an edge case and a near 30x speed improvement is worth the squeeze, but I'm just bringing context into problems I've come across in the wild and have tried to solve for in this 6.x pass